src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -0,0 +1,832 @@
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+
+  EXPECT_FALSE(rpc_service_->ManageHMICommand(message, source));
+}
+
+TEST_F(RPCServiceImplTest, ManageHMICommand_FailedCreateCommand_False) {
+  typedef am::plugin_manager::RPCPlugin RPCPlugin;
+
+  auto message = CreateMessage();
+  auto source = am::commands::Command::CommandSource::SOURCE_HMI;
+  (*message)[am::strings::params][am::strings::function_id] = kFunctionId;
+
+  ON_CALL(mock_app_mngr_, IsLowVoltage()).WillByDefault(Return(false));
+  ON_CALL(mock_app_mngr_, GetPluginManager())
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  utils::Optional<RPCPlugin> mock_rpc_plugin_opt = mock_rpc_plugin_;
+  ON_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(kFunctionId, source))
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+  ON_CALL(mock_rpc_plugin_, GetCommandFactory())
+      .WillByDefault(ReturnRef(mock_command_factory_));
+  std::shared_ptr<MockRequest> cmd;
+  ON_CALL(mock_command_factory_, CreateCommand(message, source))
+      .WillByDefault(Return(cmd));
+
+  EXPECT_FALSE(rpc_service_->ManageHMICommand(message, source));
+}
+
+TEST_F(RPCServiceImplTest, ManageHMICommand_IsAppInReconnectMode_True) {
+  typedef am::plugin_manager::RPCPlugin RPCPlugin;
+
+  auto message = CreateMessage();
+  auto source = am::commands::Command::CommandSource::SOURCE_HMI;
+  (*message)[am::strings::params][am::strings::function_id] = kFunctionId;
+  (*message)[am::strings::msg_params][am::strings::app_id] = kAppId;
+
+  ON_CALL(mock_app_mngr_, IsLowVoltage()).WillByDefault(Return(false));
+  ON_CALL(mock_app_mngr_, GetPluginManager())
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  utils::Optional<RPCPlugin> mock_rpc_plugin_opt = mock_rpc_plugin_;
+  ON_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(kFunctionId, source))
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+  ON_CALL(mock_rpc_plugin_, GetCommandFactory())
+      .WillByDefault(ReturnRef(mock_command_factory_));
+  std::shared_ptr<MockRequest> cmd =
+      std::make_shared<MockRequest>(kConnectionKey, kCorrelationId);
+  ON_CALL(mock_command_factory_, CreateCommand(message, source))
+      .WillByDefault(Return(cmd));
+
+  auto mock_app = std::make_shared<NiceMock<MockApplication> >();
+  ON_CALL(mock_app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app));
+  const connection_handler::DeviceHandle device_id1 = 1u;
+  ON_CALL(*mock_app, device()).WillByDefault(Return(device_id1));
+  ON_CALL(*mock_app, policy_app_id()).WillByDefault(Return(kPolicyAppId));
+  ON_CALL(mock_app_mngr_, IsAppInReconnectMode(device_id1, kPolicyAppId))
+      .WillByDefault(Return(true));
+
+  EXPECT_CALL(mock_command_holder_,
+              Suspend(static_cast<am::ApplicationSharedPtr>(mock_app),
+                      am::CommandHolder::CommandType::kHmiCommand,
+                      source,
+                      message))
+      .WillOnce(Return());
```

Comment:
Did you run check_style script?

Reply:
Yes, and I get such result, I need to edit it manually in addition?

Reply:
@ychernysheva **no**. If script did this then this is the way.